### PR TITLE
Configure Flix instance with `AnsiTerminalFormatter`

### LINKED ISSUE #2680: NoFormatter - no color
When running a command in VSCode (e.g. ctrl+shift+f5) I don't seem to get any syntax coloring.

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes #2680

I'm not sure the LSP Flix instance should configure a Formatter. It's guarded by an if-check but it should probably still not be there.

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/api/lsp/LanguageServer.scala [hunk 1]
```diff
@@ -289,6 +289,8 @@ class LanguageServer(port: Int) extends WebSocketServer(new InetSocketAddress("l
   private def processCheck(requestId: String)(implicit ws: WebSocket): JValue = {
     // Configure the Flix compiler.
     val flix = new Flix()
+    if (Formatter.hasColorSupport)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/LanguageServer.scala:292]: This should not use colors.

File: main/src/ca/uwaterloo/flix/language/phase/Reader.scala [hunk 2]
```diff
@@ -16,15 +16,14 @@
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Reader.scala:16]: Why these unrelated changes?

File: main/src/ca/uwaterloo/flix/Main.scala [hunk 3]
```diff
@@ -90,6 +91,24 @@ object Main {
       xstrictmono = cmdOpts.xstrictmono
     )
 
+    // configure Flix and add the paths.
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/Main.scala:94]: Why are all these lines of code moved here? I would keep them were they were.

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/Main.scala:94]: If the flix compiler is configured after the Packager is used then the implicit `Flix` instance will not be passed to the Packager

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/Main.scala:94]: I see, but now the packager sometimes creates its own Flix object and other times it doesn't. I think that is bad.

I would not pass any Flix object from main. Instead I would create a new private `newFlix` inside `Packager` and configure that correctly.

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/Main.scala:94]: Roger

File: main/src/ca/uwaterloo/flix/language/phase/Reader.scala (hunk 2 above)
Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/Reader.scala:16]: There were unused imports, so I figured I may as well clean that up but it can be reverted no problem :+1:

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Reader.scala:16]: OK, thats fine.
